From e9f9cf2ae3e4659220e69fb2392aaf6eaf25426f Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Wed, 18 Jun 2014 20:04:18 -0400
Subject: [block] blk-mq: properly drain stopped queues

Message-id: <1403121864-7956-2-git-send-email-snitzer@redhat.com>
Patchwork-id: 84263
O-Subject: [RHEL7.1 PATCH 152/150] blk-mq: properly drain stopped queues
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 8f5280f4ee75333ca12bde99ef6280ff65a8af43
Author: Christoph Hellwig <hch@lst.de>
Date:   Fri Jun 13 19:43:04 2014 +0200

    blk-mq: properly drain stopped queues

    If we need to drain a queue we need to run all queues, even if they
    are marked stopped to make sure the driver has a chance to error out
    on all queued requests.

    This fixes surprise removal with scsi-mq.

    Reported-by: Bart Van Assche <bvanassche@acm.org>
    Tested-by: Bart Van Assche <bvanassche@acm.org>

    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq.c b/block/blk-mq.c
index e11f5f8..fd8b485 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -120,7 +120,7 @@ static void __blk_mq_drain_queue(struct request_queue *q)
 
   if (count == 0)
    break;
-  blk_mq_run_queues(q, false);
+  blk_mq_start_hw_queues(q);
   msleep(10);
  }
 }
-- 
1.7.1